### PR TITLE
feat: runtime-mock and invariants

### PR DESCRIPTION
**Pull request by vgantchev**
This PR adds:
* runtime-mock - TestExternalities of the HydraDX runtime with rich data setup. This is used by the fuzzer and can be used by local (runtime integration) tests
* invariants - these are implemented as runtime panics! in try-runtime (function-level invariants) and try_state (system-level invariants)

### CODE REVIEW

File: pallets/omnipool/src/lib.rs [hunk 1]
```diff
@@ -1286,6 +1301,8 @@ pub mod pallet {
 
 			Self::update_imbalance(state_changes.delta_imbalance)?;
 
+			assert!(asset_in_state.reserve < new_asset_in_state.reserve);
```
Comment (vgantchev) [inline pallets/omnipool/src/lib.rs:1304]: @enthusiastmartin is this line kept outside of `try-runtime` feature on purpose?

File: Cargo.lock [hunk 2]
```diff
@@ -77,19 +77,19 @@ version = "0.7.7"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "5a824f2aa7e75a0c98c5a504fceb80649e9c35265d44525b5f94de4771a395cd"
 dependencies = [
- "getrandom 0.2.11",
```
Comment (vgantchev) [inline Cargo.lock:80]: @enthusiastmartin did you run `cargo update`? 

Was this necessary for something, shall we keep the version changes or revert?

Comment (vgantchev) [reply Cargo.lock]: restored old Cargo.lock to fixb uild

File: runtime-mock/src/registry.rs [hunk 3]
```diff
@@ -0,0 +1,58 @@
+use hydradx_runtime::RuntimeCall;
```
Comment (vgantchev) [inline runtime-mock/src/registry.rs:1]: Registry mock is incompatible with the new asset registry

File: pallets/omnipool/src/lib.rs (hunk 1 above)
Comment (enthusiastmartin) [reply pallets/omnipool/src/lib.rs:1304]: should be removed. i think i was testing something and left it there accidentally.

**Review by mrq1911 — APPROVED** [review]
fix tests & LGTM

**Pull request merged**